### PR TITLE
Document dependencies

### PR DESCRIPTION
**Pull request by pepicrft**
I'm adding documentation for internal and external dependencies

### CODE REVIEW

File: Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md [hunk 1]
```diff
@@ -4,11 +4,167 @@ Learn how to declare internal and external dependencies in your project.
...
+[CocoaPods](https://cocoapods.org) expects an Xcode project to integrate the dependencies. You can use Tuist to generate the project, and then run `pod install` to integrate the dependencies by creating a workspace that contains your project and the Pods dependencies. You can wrap this in a script that you can run before generating the project.
+
+```bash
+#!/usr/bin/env bash
+
+tuist generate
+pod install
+```
+
+> Warning: CocoaPods dependencies are not compatible with workflows like `build` or `test` that run `xcodebuild` right after generating the project. They are also incompatible with binary caching and selective testing since the fingerprinting logic doesn't account for the Pods dependencies.
```
Comment (danyf90) [inline Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md:77]: I would move these to the bottom and make SPM the first one

File: Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md [hunk 2]
```diff
@@ -4,11 +4,167 @@ Learn how to declare internal and external dependencies in your project.
...
+
 ### Swift Packages
 
+Swift Packages are our recommended way of declaring dependencies in your project.
+You can integrate them using Xcode's default integration mechanism or using Tuist's XcodeProj-based integration.
+
+#### Xcode's default integration
```
Comment (danyf90) [inline Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md:84]: I would swap this with the Tuist integration below

File: Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md [hunk 3]
```diff
@@ -4,11 +4,167 @@ Learn how to declare internal and external dependencies in your project.
...
+
+```swift
+let target = Target(name: "MyTarget", dependencies: [
+    .package(product: "CryptoSwift", type: .runtime)
+])
+```
+
+For Swift Macros and Build Tool Plugins, you'll need to use the types `.macro` and `.plugin` respectively.
+
+#### Tuist's XcodeProj-based integration
```
Comment (danyf90) [inline Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md:104]: I would keep this small and refer to the tutorial, WDYT?

**Review by danyf90 — APPROVED** [review]

File: Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md (hunk 3 above)
Comment (pepicrft) [reply Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md:104]: I think it's an amazing idea. I'll move it to a tutorial.

Comment (danyf90) [reply Sources/tuist/tuist.docc/Articles/Tuist/Users/dependencies.md:104]: There should already be one!

**Pull request merged**